src/background.ts

```diff
@@ -1261,22 +1271,48 @@ ipcMainHandle("HOTKEY_SETTINGS", (_, { newData }) => {
   return store.get("hotkeySettings");
 });
 
+let currentThemeMemo: ThemeConf | null = null; // ugly workaround for splash
+
 ipcMainHandle("THEME", (_, { newData }) => {
   if (newData !== undefined) {
     store.set("currentTheme", newData);
     return;
   }
+  const currentTheme = store.get("currentTheme");
   const dir = path.join(__static, "themes");
   const themes: ThemeConf[] = [];
   const files = fs.readdirSync(dir);
   files.forEach((file) => {
-    const theme = JSON.parse(fs.readFileSync(path.join(dir, file)).toString());
+    const theme = JSON.parse(
+      fs.readFileSync(path.join(dir, file)).toString()
+    ) as ThemeConf;
+    if (theme.name == currentTheme) {
+      currentThemeMemo = theme;
+    }
     themes.push(theme);
   });
-  return { currentTheme: store.get("currentTheme"), availableThemes: themes };
+  return { currentTheme, availableThemes: themes };
 });
 
 ipcMainHandle("ON_VUEX_READY", () => {
+  if (currentThemeMemo !== null) {
+    const splashURL = path.join(__static, "SplashScreen.html");
+
+    const primary = currentThemeMemo.colors.primary.substring(1);
+    const display = currentThemeMemo.colors.display.substring(1);
+    const background = currentThemeMemo.colors.background.substring(1);
+
+    const version = app.getVersion();
+
+    const fullURL = `file://${splashURL}?primary=${primary}&display=${display}&background=${background}&version=${version}`;
+
+    splash.loadURL(fullURL);
+  }
+  splash.show();
```

Comment:
I didn't notice that the current splash screen doesn't need to wait for the `vuex`. Should we also remove the `ON_VUEX_READY` call? I created it back in the `Theme` feature to prevent the main window from popping up before themes are loaded, I guess it's not necessary now as we have a splash screen.

Reply:
Indeed, `ON_VUEX_READY` may no longer be needed!